core/src/main/java/com/airbnb/aerosolve/core/transforms/DateDiffTransform.java

```diff
@@ -0,0 +1,67 @@
+package com.airbnb.aerosolve.core.transforms;
+
+
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.typesafe.config.Config;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * output = date_diff(field1, field2)
+ * get the date difference between dates in features of key "field1" and
+ * dates in features of key "field2"
+ */
+public class DateDiffTransform extends Transform {
+  private String fieldName1;
+  private String fieldName2;
+  private String outputName;
+  private final static SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+
+  @Override
+  public void configure(Config config, String key) {
+    fieldName1 = config.getString(key + ".field1");
```

Comment:
@jq I feel the BaseTransform stuffs can be done in a different PR. For this PR, it should be ok do it the current way